packages/docs/.vitepress/theme/components/MadVueBanner.vue

```diff
@@ -0,0 +1,269 @@
+                style="fill:#c4d141;fill-rule:nonzero;"></path>
+            </g>
+          </g>
+        </g>
+        <g>
+          <g>
+            <g>
+              <path
+                d="M2687.76,566.645c-0,45.92 -16.298,85.178 -48.872,117.773c-32.595,32.596 -71.853,48.894 -117.773,48.894l-28.906,-0c-45.921,-0 -85.178,-16.298 -117.774,-48.894c-32.595,-32.595 -48.871,-71.853 -48.871,-117.773l-0,-202.214c-0,-45.92 16.276,-85.178 48.871,-117.773c32.596,-32.596 71.853,-48.893 117.774,-48.893l28.906,-0c45.92,-0 85.178,16.297 117.773,48.893c32.574,32.595 48.872,71.853 48.872,117.773l-0,153.321l-251.107,-0l0,18.902c0,34.809 7.791,59.244 23.351,73.329c15.538,14.062 41.102,21.115 76.649,21.115c20.009,0 33.702,-21.484 41.124,-64.453l109.983,-0Zm-111.111,-141.102l-0,-35.547c-0,-62.24 -24.067,-93.338 -72.201,-93.338c-45.204,-0 -67.795,31.098 -67.795,93.338l0,34.44l139.996,1.107Z"
+                style="fill:#c4d141;fill-rule:nonzero;"></path>
+            </g>
+          </g>
+        </g>
+      </svg>
+    </a>
+
+    <div>
+      <div class="headline">
+        <a target="_blank" :href="target" class="vt-tagline">The
+          Vue.js Event in Madrid</a>
+        <span class="place"> · Spain</span>
+        <span class="vt-date"> · 29 May 2025</span>
+      </div>
+
+      <div class="claim">
+        <span class="mv-text-primary">Early bird</span> tickets available
+        <span class="mv-text-primary">Get 30% off</span>
+      </div>
+
+    </div>
+    <a target="_blank" class="action" :href="target">
+      More Info
+      <svg xmlns="http://www.w3.org/2000/svg" width="15" height="15" style="margin-left: 10px;" viewBox="0 0 15 15">
+        <path fill="currentColor"
+          d="M8.293 2.293a1 1 0 0 1 1.414 0l4.5 4.5a1 1 0 0 1 0 1.414l-4.5 4.5a1 1 0 0 1-1.414-1.414L11 8.5H1.5a1 1 0 0 1 0-2H11L8.293 3.707a1 1 0 0 1 0-1.414" />
+      </svg>
+    </a>
+    <div class="close-btn" @click.stop.prevent="closeBanner">
+      <span class="close">&times;</span>
+    </div>
+  </div>
+</template>
+
+
+
+<script setup lang="ts">
+import { ref, onMounted } from 'vue'
+
+const isVisible = ref(false)
+
+const nameStorage = 'MADVUE-BANNER-FEB-2025'
+
+const target = 'https://madvue.es/?utm_source=pinia&utm_content=top_banner'
+
+const resetLayoutTopHeight = () => {
+  document.documentElement.classList.add('banner-dismissed')
+}
+
+const closeBanner = () => {
+  // Hide the banner
+  isVisible.value = false
+  // Save action in the local storage
+  localStorage.setItem(nameStorage, String(true))
+  resetLayoutTopHeight()
+}
+
+onMounted(() => {
+  isVisible.value = !localStorage.getItem(nameStorage)
```

Comment:
Let's check all keys in localStorage that include `FREEWEEKEND25`. There must at least be one of them

```ts
const keys = Object.keys(localStorage).filter(key => key.includes('FREEWEEKEND25') $$ key.endsWith('_CLOSED'));

if (keys.length > 0 && keys.every(key => localStorage.getItem(key) != null)) {
    // ...
}
```